xarray/core/accessors.py

```diff
@@ -256,3 +288,39 @@ def round(self, freq):
             Array-like of datetime fields accessed for each element in values
         '''
         return self._tslib_round_accessor("round", freq)
+
+    def strftime(self, date_format):
+        '''
+        Return an array of formatted strings specified by date_format, which
+        supports the same string format as the python standard library. Details
+        of the string format can be found in `python string format doc
+        <https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior>`__
+
+        Parameters
+        ----------
+        date_format : str
+            date format string (e.g. "%Y-%m-%d")
+
+        Returns
+        -------
+        formatted strings : same type as values
+            Array-like of strings formatted for each element in values
+
+        Examples
+        --------
+        >>> rng = xr.Dataset({'time': datetime.datetime(2000, 1, 1)})
+        >>> rng['time'].dt.strftime('%B %d, %Y, %r')
+        <xarray.DataArray 'strftime' ()>
+        array('January 01, 2000, 12:00:00 AM', dtype=object)
+        """
+
+        '''
+        obj_type = type(self._obj)
+
+        result = _strftime(self._obj.data, date_format)
+
+        return obj_type(
+            result,
+            name="strftime",
```

Comment:
How do people feel about this name? Should it be the passed `date_format` instead to avoid conflicts and be explicit but also kind of messy.  Though dataarray seems like it's happy to take any string as the name.

Reply:
I wondered about this a bit too.  `'strftime'` is consistent with what we do for the other time accessors; that is, we give the resulting DataArray the name corresponding with the accessor name, even if the accessor takes an input parameter, e.g. for `round`: 
```
In [10]: da.dt.round('2D')
Out[10]:
<xarray.DataArray 'round' (dim_0: 3)>
array(['1999-12-31T00:00:00.000000000', '2000-01-02T00:00:00.000000000',
       '2000-01-04T00:00:00.000000000'], dtype='datetime64[ns]')
Coordinates:
  * dim_0    (dim_0) datetime64[ns] 2000-01-01 2000-01-02 2000-01-03
```
So I'm somewhat inclined to stick with that convention, as you've initially done here, unless others have strong opinions.

Reply:
This is fine for now, but we should consider making all of these return DataArray objects with the same name as the original object, for consistency with how xarray does other computations.